### PR TITLE
Added changes for spatial search optimization

### PR DESCRIPTION
**Pull request by omanges**
Signed-off-by: Omkar Mestry <omkar.mestry@here.com>

### CODE REVIEW

File: tests/space/test_space_objects.py [hunk 1]
```diff
@@ -600,3 +600,27 @@ def test_add_features_gpx(empty_space):
     resp = space.search(params={"p.time": "2016-06-17T23:41:13"})
     flist = list(resp)
     assert flist[0]["geometry"]["coordinates"] == [-122.391226, 37.778194, 0]
+
+
+@pytest.mark.skipif(not XYZ_TOKEN, reason="No token found.")
+def test_spatial_search_geometry_divided(large_data_space):
```
Comment (sackh) [inline tests/space/test_space_objects.py:606]: Add a docstring

File: xyzspaces/spaces.py [hunk 2]
```diff
@@ -664,20 +668,120 @@ def spatial_search_geometry(
...
             selection=selection,
             skip_cache=skip_cache,
         )
-        for f in features["features"]:
-            yield f
+
+        if features["features"]:
+            feature_list.extend(features["features"])
+
+    def add_features_geojson(
```
Comment (sackh) [inline xyzspaces/spaces.py:749]: This is a duplicate of an existing function. please remove this.

File: xyzspaces/utils.py [hunk 3]
```diff
@@ -118,8 +123,94 @@ def wkt_to_geojson(wkt_data: str) -> dict:
...
-            feature = geojson.Feature(geometry=g)
+            feature = Feature(geometry=g)
             feature_collection.append(feature)
-        return geojson.FeatureCollection(feature_collection)
+        return FeatureCollection(feature_collection)
+    else:
+        return Feature(geometry=geo)
+
+
+def grid(bbox, cellWidth, cellHeight, units):
```
Comment (sackh) [inline xyzspaces/utils.py:133]: variable names should be all small

Comment (sackh) [inline xyzspaces/utils.py:133]: Add a docstring with a description of  operation done by this function

File: xyzspaces/utils.py [hunk 4]
```diff
@@ -118,8 +123,94 @@ def wkt_to_geojson(wkt_data: str) -> dict:
...
+    bbox_height = north - south
+    columns = math.ceil(bbox_width / cell_width_deg)
+    rows = math.ceil(bbox_height / cell_height_deg)
+
+    # if the grid does not fill the bbox perfectly, center it.
+    deltaX = (bbox_width - columns * cell_width_deg) / 2
+    deltaY = (bbox_height - rows * cell_height_deg) / 2
+
+    # iterate over columns & rows
+    currentX = west + deltaX
```
Comment (sackh) [inline xyzspaces/utils.py:161]: variables should be all small letters.

File: xyzspaces/utils.py [hunk 5]
```diff
@@ -118,8 +123,94 @@ def wkt_to_geojson(wkt_data: str) -> dict:
...
+    for column in range(0, columns):
+        currentY = south + deltaY
```
Comment (sackh) [inline xyzspaces/utils.py:163]: variables should be all small letters.

File: xyzspaces/utils.py [hunk 6]
```diff
@@ -118,8 +123,94 @@ def wkt_to_geojson(wkt_data: str) -> dict:
...
+            results.append(cellPoly)
+
+            currentY += cell_height_deg
+
+        currentX += cell_width_deg
+
+    return FeatureCollection(results)
+
+
+def divide_bbox(
```
Comment (sackh) [inline xyzspaces/utils.py:190]: Add docstring with a description of this function.

File: xyzspaces/spaces.py [hunk 7]
```diff
@@ -652,10 +653,13 @@ def spatial_search_geometry(
         params: Optional[dict] = None,
         selection: Optional[List[str]] = None,
         skip_cache: Optional[bool] = None,
+        divide: Optional[bool] = False,
+        cell_width: Optional[float] = None,
+        units: Optional[str] = "m",
+        chunk_size: int = 1,
     ) -> Generator[Feature, None, None]:
         """
         Search features which intersect the provided geometry.
```
Comment (sackh) [inline xyzspaces/spaces.py:662]: add a new line

File: requirements.txt [hunk 8]
```diff
@@ -2,3 +2,4 @@
 geojson
 requests
 geopandas
+turfpy==0.0.3
```
Comment (deeplook) [inline requirements.txt:5]: Should be >=0.0.3 (?)

**Pull request merged**